test/spec/modules/dynamicAdBoostRtdProvider_spec.js

```diff
@@ -0,0 +1,69 @@
+import { subModuleObj as rtdProvider } from 'modules/dynamicAdBoostRtdProvider.js';
+import { loadExternalScript } from '../../../src/adloader.js';
+import { expect } from 'chai';
+
+const configWithParams = {
+  params: {
+    keyId: 'dynamic',
+    adUnits: ['gpt-123'],
+    threshold: 1
+  }
+};
+
+const configWithoutRequiredParams = {
+  params: {
+    keyId: ''
+  }
+};
+
+describe('dynamicAdBoost', function() {
+  describe('init', function() {
+    describe('initialize without expected params', function() {
+      it('fails initalize when keyId is not present', function() {
+        expect(rtdProvider.init(configWithoutRequiredParams)).to.be.false;
+      })
+    })
+
+    describe('initialize with expected params', function() {
+      it('successfully initialize with load script', function() {
+        expect(rtdProvider.init(configWithParams)).to.be.true;
+        setTimeout(function() {
+          expect(loadExternalScript.called).to.be.true;
+        }, 5000);
```

Comment:
We should not be using setTimeout in unit tests. This test is now effectively taking 5 seconds and may interrupt other tests and cause some weirdness.

An easier way to test this part would be to use sinon's `useFakeTimers`

```js
describe('dynamicAdBoost', function() {
  let clock;
  let sandbox;
  beforeEach(function () {
    sandbox = sinon.sandbox.create();
    clock = sandbox.useFakeTimers(Date.now());
  });
  afterEach(function () {
    sandbox.restore();
  });
```

The only problem with this is that you initialize a date time which is used to check your elapsed time is enough at module global scope, and sinon / jest do not mock timers until you tests begin to run!

So here is my proposal which I think will fix the flakiness of these tests:

1. Update your `dabStartDate` and `dabStartTime` variables to be initialized at the time of `init`
2. Use sinon's fake timers to run your tests and tick the clock instead of actually making the code wait.

Here is what I would do:

```js
// module scope declare your `dabStartDate` and `dabStartTime` vars
let dabStartDate;
let dabStartTime;

// in the init function set these to current date
function init(config, userConsent) {
   let dabStartDate = new Date();
   let dabStartTime = dabStartDate.getTime();
```


And in your test with the above mentioned initialization of fakeTimer

```
    describe('initialize with expected params', function() {
      it('successfully initialize with load script', function() {
        expect(rtdProvider.init(configWithParams)).to.be.true;
        clock.tick(1000); // so your setInterval timer is hit
        expect(loadExternalScript.called).to.be.true;
      })
    });
```

Let me know if that all makes sense.